public/app/plugins/datasource/cloudwatch/components/SQLBuilderEditor/SQLBuilderSelectRow.tsx

```diff
@@ -52,16 +53,32 @@ const SQLBuilderSelectRow: React.FC<SQLBuilderSelectRowProps> = ({ datasource, q
     [unusedDimensionKeys, schemaLabels]
   );
 
+  const onNamespaceChange = async (query: CloudWatchMetricsQuery) => {
+    const validatedQuery = await validateMetricName(query);
+    onQueryChange(validatedQuery);
+  };
+
+  const validateMetricName = async (query: CloudWatchMetricsQuery) => {
+    let { region, sql } = query;
+    await datasource.getMetrics(query.namespace, region).then((result: Array<SelectableValue<string>>) => {
+      if (!result.find((metric) => metric.value === metricName)) {
+        sql = removeMetricName(query).sql;
+      }
+    });
+    return { ...query, sql };
+  };
+
   return (
     <>
       <EditorFieldGroup>
         <EditorField label="Namespace" width={16}>
           <Select
+            aria-label="Namespace"
             value={namespace ? toOption(namespace) : null}
             inputId={`${query.refId}-cloudwatch-sql-namespace`}
             options={namespaceOptions}
             allowCustomValue
-            onChange={({ value }) => value && onQueryChange(setNamespace(query, value))}
+            onChange={({ value }) => value && onNamespaceChange(setNamespace(query, value))}
```

Comment:
new handler that validates the metric name (and resets it if not available in new namespace)